test_v2/main_test.py

```diff
@@ -0,0 +1,110 @@
+import unittest
+import pathlib
+import dataclasses
+from recipe_scrapers import scrape_html
+
+from extruct.jsonld import json
+
+__file_dir = pathlib.Path(__file__).parent.absolute()
+
+
+class RecipeTestCase(unittest.TestCase):
+    def _test_recipe(self):
+        pass
+
+
+@dataclasses.dataclass(slots=True)
+class TestCase:
+    description: str
+    expect: pathlib.Path | None
+    html: pathlib.Path | None
+
+
+@dataclasses.dataclass(slots=True)
+class TestExpectations:
+    yield_: str
+    ingredients: list[str]
+    title: str
+    author: str
+    host: str
+    canonical_url: str
+    image: str
+    total_time: int
+    instructions: str
+
+    @classmethod
+    def from_json(cls, file_path: pathlib.Path):
+        with open(file_path, "r") as json_file:
+            data = json.load(json_file)
+
+        return cls(
```

Comment:
I like that this is implemented in a way that means there would be single, standard json loader, because I think we might have to do some extra work here.

To support the recent ingredient_groups functionality, we would need to convert from the json representation to the dataclass the scraper returns. Not too difficult:
```python
...
ingredient_groups = [IngredientGroup(**group) for group in data.get("ingredient_groups", [])
...
```

There are also other functions that only some scrapers implement, like `prepTime()`, `cookTime()`, `rating()`. 
Would we have to make sure that the fields in this dataclass cover all possible functions?
Would the "expect set" only including the things that are implemented or would we have single "expect set" structure and we only populate the parts that the scraper implements?